website/docs/cdktf/concepts/providers-and-resources.html.md

```diff
@@ -264,6 +264,34 @@ export class HelloTerra extends TerraformStack {
 
 The [examples page](/docs/cdktf/examples.html) contains multiple example projects for every supported programming language.
 
+### References
+
+You can reference resource properties throughout your configuration. For example, you may want to use the name of a parent resource when assigning names to related child resources. Refer to your provider's documentation for a full list of available properties for each resource type.
+
+To create references, call `myResource.<propertyName>` on the resource instance. For example, you could use `myResource.name` to retrieve the `name` property from `myResource`. Terraform does not support passing an entire block (e.g. `exampleNamespace.metadata`) into a resource or data source, so you must create a reference for each individual property.
+
+References are also useful when you need to track logical dependencies. For example, Kubernetes resources live in a namespace, so a namespace must exist before Terraform can provision the associated resources. The TypeScript example below uses a reference for the namespace property in the the deployment. This reference tells Terraform that it needs to create the namespace before creating the resources.
```

Comment:
okay I like this a lot but now I'm feeling like people may think that if they DONT create a reference, Terraform won't know which order to create things in which isn't true...is it? I thought Terraform always knows about dependencies and creates resources accordingly. 

What am I missing here? Would we get an error if we didn't do this as a reference?

Reply:
It's true, if things are not connected through a reference it's not deterministic in which order the rest gets deployed. If one thing depends on another there needs to be a reference, otherwise it might error on deploy in some occasions (could work if you are lucky)

In HCL you use references all the time, the language is designed in a way that hints a bit more to that.